test cases/common/157 index customtarget/meson.build

```diff
@@ -24,11 +24,15 @@ gen = custom_target(
   command : [prog_python, '@INPUT@', '--header', '@OUTPUT1@', '--code', '@OUTPUT0@'],
 )
 
+assert(not is_disabler(gen), 'Custom target is a disabler.')
```

Comment:
This is bugprone. If `is_disabler` is buggy and passes a disabler object through, the entire assert will vanish. The robust way to check this is:

```meson
has_not_changed = true
if is_disabler(lib)
    has_not_changed = false
else
    has_not_changed = false
endif

assert(has_not_changed, 'message here')
```

The same goes for all these checks.

Reply:
Wouldn't that always result in an assert failure unless `is_disabler` is broken?

Reply:
Think I got something that works.